packages/blade-svelte/src/components/Link/BaseLink/BaseLink.svelte

```diff
@@ -0,0 +1,304 @@
+      xsmall: 'var(--line-height-25)',
+      small: 'var(--line-height-75)',
+      medium: 'var(--line-height-100)',
+      large: 'var(--line-height-200)',
+    },
+  };
+
+  // Reactive values based on currentInteraction - must use $derived to be reactive
+  const textDecorationLine = $derived(!isButton && currentInteraction !== 'default' ? 'underline' : 'none');
+  const iconColor = $derived(getColorToken('icon'));
+  const textColor = $derived(getColorToken('text'));
+  const iconPadding = children ? 'var(--spacing-2)' : 'var(--spacing-0)';
+  const defaultRel = target && target === '_blank' ? 'noreferrer noopener' : undefined;
+  
+  // Cursor logic: always 'pointer' unless it's a button variant AND disabled (then 'not-allowed')
+  const cursor = $derived(disabled ? 'not-allowed' : 'pointer');
+
+  // Accessibility attributes
+  const accessibilityAttrs = makeAccessible({
+    role: isButton ? 'button' : 'link',
+    disabled: isButton && isDisabled,
+    ...accessibilityProps,
+  });
+
+  // Meta attributes
+  const metaAttrs = metaAttribute({
+    name: MetaConstants.Link,
+    testID,
+  });
+
+  // Analytics attributes
+  const analyticsAttrs = makeAnalyticsAttribute(rest);
+
+  // Event handlers
+  function handleClick(event: MouseEvent): void {
+    if (onClick && !disabled) {
+      onClick(event);
+    }
+  }
+
+  function handleFocus(event: FocusEvent): void {
+    if (!disabled) {
+      handleFocusInteraction();
+    }
+    onFocus?.(event);
+  }
+
+  function handleBlur(event: FocusEvent): void {
+    if (!disabled) {
+      handleBlurInteraction();
+    }
+    onBlur?.(event);
+  }
+
+  function handleMouseEnter(): void {
+    if (!disabled) {
+      handleMouseEnterInteraction();
+    }
+  }
+
+  function handleMouseLeave(event: MouseEvent): void {
+    if (!disabled) {
+      handleMouseLeaveInteraction();
+    }
+    onMouseLeave?.(event);
+  }
+
+  // Computed styles
+  const linkClass = `base-link base-link--${variant} base-link--${color} base-link--${size}`;
+  const contentClass = 'base-link__content';
+  const textClass = `base-link__text base-link__text--${size}`;
+</script>
+
+{#if isButton}
+  <button
+    class={linkClass}
+    class:disabled={disabled}
+    type="button"
+    disabled={disabled}
+    title={htmlTitle}
+    style="opacity: {opacity ?? 1}; cursor: {cursor};"
+    {...accessibilityAttrs}
+    {...metaAttrs}
+    {...analyticsAttrs}
+    onclick={handleClick}
+    onfocus={handleFocus}
+    onblur={handleBlur}
+    onmouseenter={handleMouseEnter}
+    onmouseleave={handleMouseLeave}
+    onmousemove={onMouseMove}
+    onpointerdown={onPointerDown}
+    onpointerenter={onPointerEnter}
+    ontouchstart={onTouchStart}
+    ontouchend={onTouchEnd}
+    onmousedown={onMouseDown}
+    onmouseup={onMouseUp}
+  >
+    <span class={contentClass}>
+      {#if Icon && iconPosition === 'left'}
+        <span class="base-link__icon base-link__icon--left" style="padding-right: {iconPadding}; color: {iconColor};">
```

Comment:
do we have any other options apart from inline styling for such scenarios? can you check if CVA link I shared above helps in these scenarios? 

inline styling has a bit too much of specificity and we won't be able to add media queries or hover stylings this way if we have dynamic colors defined in js 🤔